csvkit/utilities/csvsql.py

```diff
@@ -48,12 +46,14 @@ def main(self):
         do_insert = self.args.insert
         query = self.args.query
 
-        if self.args.queryfile:
-            if self.args.queryfile == '-':
-                queryfile = sys.stdin
-            else:
-                queryfile = open(self.args.queryfile, 'U')
-            query = queryfile.read().replace('\n', ' ')
+        # support reading query string from a file if the argument exists on the filesystem
+        if query == '-':
+            queryfile = sys.stdin
+        elif os.path.exists(query):
```

Comment:
Not using isfile here since we (or I) want FIFOs and links too, but if exists is too general I can switch this to stat S_IS* checks.